### PR TITLE
Simplify generation of arbitrary functions.

### PR DESCRIPTION
**Pull request by jonathanknowles**
## Issue

None. (Noticed while reviewing code.)

## Description

This PR simplifies the generation of arbitrary functions within `cardano-wallet-primitive` and `cardano-coin-selection`.

### CODE REVIEW

File: lib/primitive/test/spec/Cardano/Wallet/Primitive/Types/UTxOSpec.hs [hunk 1]
```diff
@@ -402,8 +399,7 @@ deriving anyclass instance Function AssetName
 deriving anyclass instance CoArbitrary TokenPolicyId
 deriving anyclass instance Function TokenPolicyId
 
-instance CoArbitrary TxIn where
-    coarbitrary = coarbitraryTxIn
+deriving anyclass instance CoArbitrary TxIn
```
Comment (jonathanknowles) [inline lib/primitive/test/spec/Cardano/Wallet/Primitive/Types/UTxOSpec.hs:402]: Note that this relies on the default implementation of `coarbitrary`, available for types that are instances of `Generic`. (See https://hackage.haskell.org/package/QuickCheck-2.14.3/docs/Test-QuickCheck.html#v:coarbitrary)

**Review by HeinrichApfelmus — APPROVED** [review]
Looks good to me, thanks! 😊

File: lib/primitive/test/spec/Cardano/Wallet/Primitive/Types/UTxOSpec.hs (hunk 1 above)
Comment (paweljakubas) [reply lib/primitive/test/spec/Cardano/Wallet/Primitive/Types/UTxOSpec.hs:402]: :+1:

**Review by paweljakubas — APPROVED** [review]
nice catch and refactoring

**Pull request merged**